src/countFactorials.cpp

```diff
@@ -32,22 +32,24 @@ long fact(int num)
 		fact *= i;
 	}
 	return fact;
-}
+}*/
 int countFactorials(long l, long h)
 {
 	if ((l < 0 || h < 0) || l > h)
 	{
 		return -1;
 	}
 	int count=0;
-	long i=0;
+	long i=1,f=1;
 	while(1)
 	{
-		if(fact(i)>=l && fact(i)<=h)
+		if(f>=l && f<=h)
 		count++;
-		if (fact(i) > h)
+		if (f > h)
 			break;
+		f = f*i;
 		i++;
+		
```

Comment:
Try something like isFactorial. Not necessarily now but you can refer to it in future if you have time.